src/core/qgsgpkgflusher.h

```diff
@@ -0,0 +1,55 @@
+/***************************************************************************
+                          qgsgpkgflusher.h
+                             -------------------
+  begin                : Oct 2019
+  copyright            : (C) 2019 by Matthias Kuhn
+  email                : matthias@opengis.ch
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSGPKGFLUSHER_H
+#define QGSGPKGFLUSHER_H
+
+#include <QObject>
+#include <qgsmaplayer.h>
+
+class QgsProject;
+class Flusher;
+
+/**
+ * The QgsGpkgFlusher class is attached to a QgsProject.
+ * It will make sure that all changes are regularly flushed from the wal file
+ * to the gpkg itself on all added layers.
```

Comment:
```suggestion
 * to the gpkg itself on all added layers.
 * It will start a background thread and post an event to it whenever the gpkg has been changed.
 * After a delay of 500ms without any changes the wal file will be flushed.
```